src/ast/irbuilderbpf.cpp

```diff
@@ -621,7 +621,13 @@ void IRBuilderBPF::CreatePerfEventOutput(Value *ctx, Value *data, size_t size)
   // int bpf_perf_event_output(ctx, map, flags, data, size)
   FunctionType *perfoutput_func_type = FunctionType::get(
       getInt64Ty(),
-      {getInt8PtrTy(), getInt8PtrTy(), getInt64Ty(), getInt8PtrTy(), getInt64Ty()},
+      {
+        getInt8PtrTy(),
+        getInt64Ty(),
```

Comment:
`int bpf_perf_event_output(struct pt_reg *ctx, struct bpf_map *map, u64 flags, void *data, u64 size)`
This argument is meant to be a pointer

Reply:
Yep, but the argument we pass is always of type i64. Return type of llvm.bpf.pseudo is i64 and this is what we pass to bpf_perf_event_output. Should I add one more change where we'll define an opaque type "bpf_map" or something like that and cast results of llvm.bpf.pseudo to bpf_map* ?